File: CodingInterviews/LinkList/ReverseList.java
package LinkList;

// leetcode 206
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode p = new ListNode(-1);
        while (head != null) {
            ListNode t = head;
            head = head.next;
            t.next = p.next;
            p.next = t;
        }
        return p.next;
    }
}

File: CodingInterviews/LinkList/ReverseList.cpp
#include "listnode.h"

class Solution {
  public:
    ListNode *ReverseList(ListNode *pHead) {
        ListNode *head = new ListNode(-1);
        ListNode *p = pHead;
        while (p != nullptr) {
            ListNode *t = new ListNode(p->val);
            t->next = head->next;
            head->next = t;
            p = p->next;
        }
        p = head;
        head = head->next;
        delete p;
        return head;
    }
};